pkg/idempotent/redis/redis.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+// redis is an immemory Recorder implementation.
+
+package redis
+
+import (
+	"context"
+	"os"
+	"time"
+
+	redis "github.com/gomodule/redigo/redis"
+)
+
+var (
+	// Used for Redis value
+	defaultAddress = "localhost:6379"
+	defaultTimeout = 86400 //The default timeout should be 24 hours.
+	defaultValue   = struct{}{}
+)
+
+func New() *Redis {
+	return &Redis{}
+}
+
+type Redis struct {
+}
+
+func (r *Redis) SeenBefore(key string) bool {
+	ctx, _ := context.WithTimeout(context.TODO(), 25*time.Millisecond)
+	if addr := os.Getenv("REDIS_INSTANCE"); addr != "" {
+		defaultAddress = addr
+	}
+	conn, err := redis.Dial("tcp", defaultAddress)
+	if err != nil {
+		ctx = context.WithValue(ctx, "redis dial error", err)
```

Comment:
FYI, often anonymous structs are used for context key's. They're easier than repeating a string and guaranteed to be unique (which is good if you're passing a `Context` off to another package).

https://github.com/mitchellh/go-server-timing/blob/master/context.go